baybe/exceptions.py

```diff
@@ -10,6 +10,19 @@ class UnusedObjectWarning(UserWarning):
 
 
 ##### Exceptions #####
+
+
+class IncompatibilityError(Exception):
```

Comment:
either make this incompatability error more specific taylored to the suage
OR
remove the specific incompatability error from below and inplace use this generic one here

Reply:
Maybe should have clarified my vision here:

With the upcoming "validation mechanism", we'll probably need to handle a variety of different "incompatibility errors" in different contexts (e.g. target being incompatible with surrogate, recommender being incompatible with constraint, ...). It makes sense to group all of them under a common base class (like `IncompatibilityError`) so that you can collectively catch them with one single `except IncompatibilityError` expression but still be able to fine-select if you need to. Since here we have the first situation where I need to catch such an error that is different from the already existing `IncompatibleSearchSpaceError`, I introduced this base class already now. If you want, I can create a subclass for the situation required in this PR.

And probably, we should have done this much earlier with all our other exceptions as well – we can start to slowly migrate this in the next weeks. For example, all existing baybe errors that essentially express a `ValueError` (e.g. `InvalidTargetValueError`) could actually subclass python's value error, to allow for the same handling.

Does this make sense to you?